lib/ecmascript.ts

```diff
@@ -1180,56 +1201,88 @@ type AllKeys<U> = Keys<U extends unknown ? U : never>;
 type AnyTemporalKey = Exclude<AllKeys<AnyTemporalLikeType>, symbol>;
 type Keys<T> = T extends Record<string, unknown> ? keyof T : never;
 type Values<T> = T[keyof T];
-type Entry<K> = K | readonly [K, (string | number | undefined)?];
-type ResolvedFieldTypes<T> = T extends Entry<infer K> ? K : never;
 type Owner<T extends AnyTemporalKey> = Values<{
```

Comment:
I think this could maybe be simplified to

```typescript
type Owner<K extends AnyTemporalKey> = K extends unknown ? OwnerOf<K, AnyTemporalLikeType> : never;
type OwnerOf<K extends AnyTemporalKey, T> = T extends Record<K, unknown> ? T : never;
```

At which point maybe you don't even need `AllTemporalLikeTypes` or `Values` at all anymore?

Reply:
That was very helpful! I worked this in, though because all the TemporalLike types have optional properties I needed to make some adjustments to get `T extends Record<K...>` to work. PTAL.